### PR TITLE
Create a pattern for drilling down to a single "show objects like" row

### PR DESCRIPTION
**Pull request by sfc-gh-cgorrie**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Introduced `SqlExecutionMixin.show_specific_object` to share code between all the times we need to `show objects like` in order to get more information about a particular entity.

Original author: @sfc-gh-davwang

### CODE REVIEW

File: src/snowflake/cli/plugins/nativeapp/manager.py
Comment (sfc-gh-bgoel) [inline src/snowflake/cli/plugins/nativeapp/manager.py:102]: What is the benefit to making a new Mixin as opposed to introducing the main method `find_row_by_object_name` in `SqlExecutionMixin` ?  New managers would only need to inherit from the`SqlExecutionMixin`, and we do not need to remember this additional class. It is possible that those who forget to include this new class when defining a new manager end up creating their own such method, which could be avoided if it were already contained in `SqlExecutionMixin`

Comment (sfc-gh-davwang) [reply src/snowflake/cli/plugins/nativeapp/manager.py:102]: FWIW, I think that if this can be included in SqlExecutionMixin then it would be most useful in there. I just wasn't initially sure whether SqlExecutionMixin is meant to be, as Cam says, "lower-level than these sort of helpers."

Comment (sfc-gh-bgoel) [reply src/snowflake/cli/plugins/nativeapp/manager.py:102]: I see your point @sfc-gh-cgorrie and it makes sense. Will leave it up to you, happy to approve once you have decided however you want to proceed.

File: src/snowflake/cli/plugins/nativeapp/run_processor.py
Comment (sfc-gh-davwang) [inline src/snowflake/cli/plugins/nativeapp/run_processor.py:234]: nit: could be shortened to `return self.show_specific_object(...)`, no need to wait until after the except block

File: tests/nativeapp/test_manager.py
Comment (sfc-gh-davwang) [inline tests/nativeapp/test_manager.py:17]: are Mock and SqlExecutionMixin imports used in this test file?

**Review by sfc-gh-davwang — APPROVED** [review]

File: src/snowflake/cli/plugins/nativeapp/run_processor.py
Comment (sfc-gh-cgorrie) [reply src/snowflake/cli/plugins/nativeapp/run_processor.py:234]: You'd think so, but sadly the linter complains that there's no return (because the except block calls the generic error handler). I could add an explicit `return None` there, might feel cleaner?

Comment (sfc-gh-davwang) [reply src/snowflake/cli/plugins/nativeapp/run_processor.py:234]: ah you're right about the except block, this probably works best then

**Comment by sfc-gh-davwang** [comment]
if you're amenable to including it, here's a PR to use your new function in image_registry/manager: https://github.com/Snowflake-Labs/snowcli/pull/755

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-bgoel — APPROVED** [review]
LGTM for both Project and NA

File: src/snowflake/cli/api/sql_execution.py
Comment (sfc-gh-cgorrie) [reply src/snowflake/cli/api/sql_execution.py:169]: To me, that's a different layer. My mental model of layers:

- command "spec" / app layer (i.e. `commands.py`)
- command implementation layer (i.e. managers; e.g. `ObjectManager.py`)
- utility layer (e.g. `utils/*.py`, and `SqlExecutionMixin`)

Suppose we were to put it in `ObjectManager`. Other managers that want to use it would have to instantiate an `ObjectManager()`, then call this. Right now there's no args for the constructor but I'm not sure we can count on that going forward given the design (e.g. if we ever needed to support multiple concurrent connections). So this felt like the cleanest place to put it.

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**
